graphql_compiler/macros/__init__.py

```diff
@@ -104,7 +122,25 @@ def get_schema_with_macros(macro_registry):
     Returns:
         GraphQLSchema with additional fields where macroe edges can be used.
     """
-    raise NotImplementedError()  # TODO(bojanserafimov) implement
+    schema_ast = parse(print_schema(macro_registry.schema_without_macros))
+    for definition in schema_ast.definitions:
+        if isinstance(definition, ObjectTypeDefinition):
+            if definition.name.value in macro_registry.macro_edges:
+                macro_edges = macro_registry.macro_edges[definition.name.value]
+                for macro_edge_name, macro_edge_descriptor in six.iteritems(macro_edges):
+                    type_at_target = get_type_at_macro_edge_target(
+                        macro_registry.schema_without_macros,
+                        macro_edge_descriptor.expansion_ast)
+                    arguments = []
+                    definition.fields.append(FieldDefinition(
+                        Name(macro_edge_name), arguments,
+                        NamedType(Name(type_at_target.name)), directives=[]))
+    return build_ast_schema(schema_ast)
+
+
+def get_schema_for_macro_definition(schema):
+    """Returns a schema with macro directives."""
```

Comment:
Can you elaborate more on the spec of this function? I don't see the link between the docstring and the function name.